### ISSUE TITLE
[Bug] Output Heads Degenerate via Linear Collapse (Chained Linear Layers With Zero Activations)

### ISSUE THREAD

**Issue by RustamSheoran** [Mohammedmt10/self-driving-model-in-simulation#8]
## Labels
`bug`, `model-architecture`, `high`

## Affected Files
- `model.py` (lines 106–128)

## Problem Summary
In `steering_head`, `throttle_head`, and `brake_head`, four `nn.Linear` layers are stacked consecutively with no non-linear activation functions between them.

## Root Cause Code
```python
# model.py lines 106-128
self.steering_head = nn.Sequential(
    nn.Linear(128, 64),
    nn.Linear(64, 32),
    nn.Linear(32, 16),
    nn.Linear(16, 1),
    nn.Tanh(),
)

self.throttle_head = nn.Sequential(
    nn.Linear(128, 64),
    nn.Linear(64, 32),
    nn.Linear(32, 16),
    nn.Linear(16, 1),
    nn.Sigmoid(),
)

self.brake_head = nn.Sequential(
    nn.Linear(128, 64),
    nn.Linear(64, 32),
    nn.Linear(32, 16),
    nn.Linear(16, 2),
)
```

## Why It Is Completely Ruining Everything
In linear algebra, any product of matrix multiplications without non-linear activations collapses into a single affine transformation:
$$W_{\text{total}} = W_4 \cdot W_3 \cdot W_2 \cdot W_1$$
This 4-layer chain has **identically the same capacity as a single linear layer** (`nn.Linear(128, 1)`). Furthermore, forcing the features through a 16-dimensional bottleneck prior to output artificially restricts rank, increases parameter bloat, and harms backpropagation gradient flow without providing any non-linear capacity to model complex driving behaviors.

## Proposed Fix
Insert non-linear activations between layers or simplify to a clean 2-layer MLP:
```python
self.steering_head = nn.Sequential(
    nn.Linear(128, 64),
    nn.ReLU(),
    nn.Linear(64, 1),
    nn.Tanh(),
)

self.throttle_head = nn.Sequential(
    nn.Linear(128, 64),
    nn.ReLU(),
    nn.Linear(64, 1),
    nn.Sigmoid(),
)

self.brake_head = nn.Sequential(
    nn.Linear(128, 64),
    nn.ReLU(),
    nn.Linear(64, 2),
)
```
